src/reanimated2/layoutReanimation/animationBuilder/Keyframe.ts

```diff
@@ -0,0 +1,199 @@
+import { Easing, EasingFn } from '../../Easing';
+import { withDelay, withSequence, withTiming } from '../../animations';
+import {
+  AnimationFunction,
+  EntryExitAnimationBuild,
+  IEntryExitAnimationBuilder,
+  KeyFrameProps,
+  StyleProps,
+} from './commonTypes';
+
+export interface KeyFramePoint {
+  duration: number;
+  value: number | string;
+  easing?: EasingFn;
+}
+export interface ParsedKeyFramesDefinition {
+  initialValues: StyleProps;
+  keyframes: {
+    [key: string]: KeyFramePoint[];
+  };
+}
+
+export interface KeyFramesDefinition {
+  [key: number]: KeyFrameProps;
+}
+export class Keyframe implements IEntryExitAnimationBuilder {
+  durationV?: number;
+  delayV?: number;
+  definitions: KeyFramesDefinition;
+
+  /*
+    Keyframe definition should be passed in the constructor as the map
+    which keys are between range 0 - 100 (%) and correspond to the point in the animation progress.
+  */
+  constructor(definitions: KeyFramesDefinition) {
+    this.definitions = definitions;
+  }
+
+  private parseDefinitions(): ParsedKeyFramesDefinition {
+    /* 
+        for each style property contains an array with all its key points: 
```

Comment:
I don't understand this comment, should it read "Each style property contain..."?